### PR TITLE
Used toHeapData when converting native-data --> heap-data

### PR DESCRIPTION
**Pull request by ahmetmircik**

### CODE REVIEW

**Comment by ahmetmircik** [comment]
`com.hazelcast.client.quorum.cache.ClientCacheWriteQuorumTest.testRemoveAllOperationSuccessfulWhenQuorumSizeMet`

```
com.hazelcast.core.HazelcastException: java.lang.AssertionError: CacheService[hz:impl:cacheService] is instance of FragmentedMigrationAwareService, com.hazelcast.cache.impl.operation.CacheRemoveAllOperation{serviceName='hz:impl:cacheService', identityHash=662590027, partitionId=31, replicaIndex=0, callId=158, invocationTime=1505386624623 (2017-09-14 10:57:04.623), waitTimeout=-1, callTimeout=60000, name=/hz/cacheQuorum2921314d-859e-4b8e-8bc9-d48bc8c64230} should implement ServiceNamespaceAware!
	at com.hazelcast.util.ExceptionUtil.peel(ExceptionUtil.java:94)
	at com.hazelcast.util.ExceptionUtil.peel(ExceptionUtil.java:56)
	at com.hazelcast.util.ExceptionUtil.peel(ExceptionUtil.java:52)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.sendClientMessage(AbstractMessageTask.java:221)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.handleProcessingFailure(AbstractMessageTask.java:162)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.run(AbstractMessageTask.java:107)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
	at ------ submitted from ------.(Unknown Source)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.resolveAndThrowIfException(ClientInvocationFuture.java:96)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.resolveAndThrowIfException(ClientInvocationFuture.java:33)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:155)
	at com.hazelcast.client.cache.impl.AbstractClientCacheProxyBase.invoke(AbstractClientCacheProxyBase.java:208)
	at com.hazelcast.client.cache.impl.AbstractClientInternalCacheProxy.removeAllInternal(AbstractClientInternalCacheProxy.java:540)
	at com.hazelcast.client.cache.impl.ClientCacheProxy.removeAll(ClientCacheProxy.java:284)
```

**Comment by ahmetmircik** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/AbstractMapDataStore.java [hunk 1]
```diff
@@ -77,8 +78,8 @@ protected Object toObject(Object obj) {
         return serializationService.toObject(obj);
     }
 
-    protected Data toData(Object obj) {
-        return serializationService.toData(obj);
+    protected Data toHeapData(Object obj) {
+        return ToHeapDataConverter.toHeapData(serializationService.toData(obj));
```
Comment (mdogan) [inline hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/AbstractMapDataStore.java]: What is the advantage of `ToHeapDataConverter `?

File: hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/AbstractMapDataStore.java [hunk 2]
```diff
@@ -16,6 +16,7 @@
 
 package com.hazelcast.map.impl.mapstore;
 
+import com.hazelcast.internal.nearcache.impl.invalidation.ToHeapDataConverter;
```
Comment (mdogan) [inline hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/AbstractMapDataStore.java]: Importing a nearcache internal class to mapstore package... That looks odd.

File: hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/AbstractMapDataStore.java (hunk 1 above)
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/AbstractMapDataStore.java]: fixed

File: hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/AbstractMapDataStore.java (hunk 2 above)
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/AbstractMapDataStore.java]: fixed

**Comment by mmedenjak** [comment]
@ahmetmircik https://github.com/hazelcast/hazelcast/pull/11366

**Pull request merged**